cni/pkg/plugin/plugin.go

```diff
@@ -224,7 +224,8 @@ func CmdAdd(args *skel.CmdArgs) (err error) {
 		log.Infof("ambientConf.Mode: %s", ambientConf.Mode)
 		log.Infof("ambientConf.ZTunnelReady: %v", ambientConf.ZTunnelReady)
 		added := false
-		if !excludePod && ambientConf.Mode != ambient.AmbientMeshOff.String() && ambientConf.ZTunnelReady {
+		// TODO: RedirectMode should be defined in api proto
+		if ambientConf.RedirectMode != "ebpf" && !excludePod && ambientConf.Mode != ambient.AmbientMeshOff.String() && ambientConf.ZTunnelReady {
```

Comment:
I didn't see the rationale in the proposal - why are we eschewing the existing plugin model here?

e.g. why

`istio-iptables` -> CLI plugin the CNI layer invokes on events.
`cni server` > webserver running in the CNI layer that is invoked on events.

and not

`istio-iptables` -> CLI plugin the CNI layer invokes on events.
`istio-ebpf` -> CLI plugin the CNI layer invokes on events.

Reply:
Okay I misunderstood - there's just the one iptables CNI plugin and we if-condition out of it and start a local server for eBPF ops.

I still don't like the internal embedded server approach tho, it means you have to rebuild the entire CNI plugin to modify/swap out the eBPF progs, or ship your own alternate entire iptables + eBPF Istio CNI plugin, even if you only cared to tweak one or the other.

Reply:
there is no webserver for ebpf  in CNI

Reply:
No webserver, but there is a `server` Goroutine in `cni/pkg/ebpf/server/redirectionServer.go` that is started when the CNI plugin starts, and hardcodes the eBPF progs, correct?

Reply:
Aight - I think how we load progs and maps and whether they should be baked into the CNI plugin or optionally loadable via path/volume mount is an open question that is it's own separate, larger discussion, and any changes that come out of that would be a followup and don't need to block this specific PR.